File: generated-code/codecraft/javascript/TcpReadWrite/model/entity.js
const EntityType = require('./entity-type');
const Vec2Int = require('./vec2-int');
/**
 * Game entity
 */
class Entity {
    /**
     * Entity's ID. Unique for each entity
     */
    id;
    /**
     * Entity's owner player ID, if owned by a player
     */
    playerId;
    /**
     * Entity's type
     */
    entityType;
    /**
     * Entity's position (corner with minimal coordinates)
     */
    position;
    /**
     * Current health
     */
    health;
    /**
     * If entity is active, it can perform actions
     */
    active;

    constructor(id, playerId, entityType, position, health, active) {
        this.id = id;
        this.playerId = playerId;
        this.entityType = entityType;
        this.position = position;
        this.health = health;
        this.active = active;
    }

    /**
     * Read Entity from input stream
     */
    static async readFrom(stream) {
        let id;
        id = await stream.readInt();
        let playerId;
        if (await stream.readBool()) {
            playerId = await stream.readInt();
        } else {
            playerId = null;
        }
        let entityType;
        entityType = await EntityType.readFrom(stream);
        let position;
        position = await Vec2Int.readFrom(stream);
        let health;
        health = await stream.readInt();
        let active;
        active = await stream.readBool();
        return new Entity(id, playerId, entityType, position, health, active);
    }

    /**
     * Write Entity to output stream
     */
    async writeTo(stream) {
        let id = this.id;
        await stream.writeInt(id);
        let playerId = this.playerId;
        if (playerId === null) {
            await stream.writeBool(false);
        } else {
            await stream.writeBool(true);
            await stream.writeInt(playerId);
        }
        let entityType = this.entityType;
        await entityType.writeTo(stream);
        let position = this.position;
        await position.writeTo(stream);
        let health = this.health;
        await stream.writeInt(health);
        let active = this.active;
        await stream.writeBool(active);
    }
}
module.exports = Entity

File: generated-code/codecraft/kotlin/TcpReadWrite/src/main/kotlin/model/Entity.kt
package model

import util.StreamUtil

/**
 * Game entity
 */
class Entity {
    /**
     * Entity's ID. Unique for each entity
     */
    var id: Int = 0
    /**
     * Entity's owner player ID, if owned by a player
     */
    var playerId: Int? = null
    /**
     * Entity's type
     */
    lateinit var entityType: model.EntityType
    /**
     * Entity's position (corner with minimal coordinates)
     */
    lateinit var position: model.Vec2Int
    /**
     * Current health
     */
    var health: Int = 0
    /**
     * If entity is active, it can perform actions
     */
    var active: Boolean = false

    constructor(id: Int, playerId: Int?, entityType: model.EntityType, position: model.Vec2Int, health: Int, active: Boolean) {
        this.id = id
        this.playerId = playerId
        this.entityType = entityType
        this.position = position
        this.health = health
        this.active = active
    }

    /**
     * Write Entity to output stream
     */
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, id)
        val playerIdValue = playerId
        if (playerIdValue == null) {
            StreamUtil.writeBoolean(stream, false)
        } else {
            StreamUtil.writeBoolean(stream, true)
            StreamUtil.writeInt(stream, playerIdValue)
        }
        StreamUtil.writeInt(stream, entityType.tag)
        position.writeTo(stream)
        StreamUtil.writeInt(stream, health)
        StreamUtil.writeBoolean(stream, active)
    }

    /**
     * Get string representation of Entity
     */
    override fun toString(): String {
        var stringBuilder = StringBuilder("Entity { ")
        stringBuilder.append("id: ")
        stringBuilder.append(id)
        stringBuilder.append(", ")
        stringBuilder.append("playerId: ")
        stringBuilder.append(playerId)
        stringBuilder.append(", ")
        stringBuilder.append("entityType: ")
        stringBuilder.append(entityType)
        stringBuilder.append(", ")
        stringBuilder.append("position: ")
        stringBuilder.append(position)
        stringBuilder.append(", ")
        stringBuilder.append("health: ")
        stringBuilder.append(health)
        stringBuilder.append(", ")
        stringBuilder.append("active: ")
        stringBuilder.append(active)
        stringBuilder.append(" }")
        return stringBuilder.toString()
    }

    companion object {
        /**
         * Read Entity from input stream
         */
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): Entity {
            var id: Int
            id = StreamUtil.readInt(stream)
            var playerId: Int?
            if (StreamUtil.readBoolean(stream)) {
                playerId = StreamUtil.readInt(stream)
            } else {
                playerId = null
            }
            var entityType: model.EntityType
            entityType = model.EntityType.readFrom(stream)
            var position: model.Vec2Int
            position = model.Vec2Int.readFrom(stream)
            var health: Int
            health = StreamUtil.readInt(stream)
            var active: Boolean
            active = StreamUtil.readBoolean(stream)
            return Entity(id, playerId, entityType, position, health, active)
        }
    }
}